// contentful-management/dist/es-modules/common-utils.js
const _excluded = ["pagePrev", "pageNext"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
/* eslint-disable @typescript-eslint/ban-ts-comment */

import { toPlainObject } from 'contentful-sdk-core';
import copy from 'fast-copy';
/**
 * @private
 */
export const wrapCollection = fn => (makeRequest, data, ...rest) => {
  const collectionData = toPlainObject(copy(data));
  // @ts-expect-error
  collectionData.items = collectionData.items.map(entity => fn(makeRequest, entity, ...rest));
  // @ts-expect-error
  return collectionData;
};
export const wrapCursorPaginatedCollection = fn => (makeRequest, data, ...rest) => {
  const collectionData = toPlainObject(copy(data));
  // @ts-expect-error
  collectionData.items = collectionData.items.map(entity => fn(makeRequest, entity, ...rest));
  // @ts-expect-error
  return collectionData;
};
export function isSuccessful(statusCode) {
  return statusCode < 300;
}
export function shouldRePoll(statusCode) {
  return [404, 422, 429, 400].includes(statusCode);
}
export async function waitFor(ms = 1000) {
  return new Promise(resolve => setTimeout(resolve, ms));
}
export function normalizeCursorPaginationParameters(query) {
  const {
      pagePrev,
      pageNext
    } = query,
    rest = _objectWithoutProperties(query, _excluded);
  return _objectSpread(_objectSpread(_objectSpread({}, rest), {}, {
    cursor: true
  }, pagePrev ? {
    pagePrev
  } : null), pageNext ? {
    pageNext
  } : null);
}
function extractQueryParam(key, url) {
  var _URLSearchParams$get;
  if (!url) return;
  const queryIndex = url.indexOf('?');
  if (queryIndex === -1) return;
  const queryString = url.slice(queryIndex + 1);
  return (_URLSearchParams$get = new URLSearchParams(queryString).get(key)) !== null && _URLSearchParams$get !== void 0 ? _URLSearchParams$get : undefined;
}
const Pages = {
  prev: 'pagePrev',
  next: 'pageNext'
};
const PAGE_KEYS = ['prev', 'next'];
export function normalizeCursorPaginationResponse(data) {
  const pages = {};
  for (const key of PAGE_KEYS) {
    var _data$pages;
    const token = extractQueryParam(Pages[key], (_data$pages = data.pages) === null || _data$pages === void 0 ? void 0 : _data$pages[key]);
    if (token) pages[key] = token;
  }
  return _objectSpread(_objectSpread({}, data), {}, {
    pages
  });
}
