// contentful-management/dist/es-modules/adapters/REST/endpoints/bulk-action.js
import * as raw from './raw';
export const get = (http, params) => {
  return raw.get(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions/actions/${params.bulkActionId}`);
};
export const publish = (http, params, payload) => {
  return raw.post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions/publish`, payload);
};
export const unpublish = (http, params, payload) => {
  return raw.post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions/unpublish`, payload);
};
export const validate = (http, params, payload) => {
  return raw.post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions/validate`, payload);
};
export const getV2 = (http, params) => {
  return raw.get(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions/${params.bulkActionId}`);
};
export const publishV2 = (http, params, payload) => {
  return raw.post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions`, payload);
};
export const unpublishV2 = (http, params, payload) => {
  return raw.post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions`, payload);
};
export const validateV2 = (http, params, payload) => {
  return raw.post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions`, payload);
};
